### PR TITLE
Adding destructors and delete functions for clearing up the memory

### PR DESCRIPTION
**Pull request by Durganshu**
This PR adds functions for deleting a component, its subtree (all the children), and all the associated data paths. Any component can call these functions, and they serve in deallocating the memory heap.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.22)
 
 set (CMAKE_CXX_STANDARD 20)
 set (CXX_STANDARD_REQUIRED ON)
-set (CMAKE_CXX_FLAGS " -Wall")
+set (CMAKE_CXX_FLAGS " -Wall -ggdb3")
```
Comment (stepanvanecek) [inline CMakeLists.txt:6]: is this a leftover from your debugging or did you leave it on purpose?

File: src/Topology.cpp [hunk 2]
```diff
@@ -354,9 +362,71 @@ int Component::GetTopologySize(unsigned * out_component_size, unsigned * out_dat
...
+    }    
+    return;
+}
+void Component::Delete(bool withSubtree)
+{
+    // Delete subtree and all data paths
+    if (withSubtree)
+    {
+        DeleteSubtree();
+        DeleteAllDataPaths();
```
Comment (stepanvanecek) [inline src/Topology.cpp:398]: this gets executed in both if+else -> you can move it out of there

**Review by stepanvanecek — APPROVED** [review]
please resolve the smaller comments, otherwise looks good

**Comment by stepanvanecek** [comment]
oh, one more thing, can you direct your PRs to the develop branch the next time?

**Comment by Durganshu** [comment]
> oh, one more thing, can you direct your PRs to the develop branch the next time?

Yes, sure. Will ensure this in future PRs

**Pull request merged**